### ISSUE TITLE
BulkWithUpdatesIT.testFailedRequestsOnClosedIndex fails

### ISSUE THREAD

**Issue by tlrx** [elastic/elasticsearch#37571]
The test `BulkWithUpdatesIT.testFailedRequestsOnClosedIndex` failed on CI:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+periodic/521

It does not reproduce locally:
```
./gradlew :server:integTest -Dtests.seed=B6EBD34952F53489 -Dtests.class=org.elasticsearch.action.bulk.BulkWithUpdatesIT -Dtests.method="testFailedRequestsOnClosedIndex" -Dtests.security.manager=true -Dtests.locale=cs -Dtests.timezone=Africa/Djibouti -Dcompiler.java=11 -Druntime.java=8
```

The test execution log is : [consoleText.txt](https://github.com/elastic/elasticsearch/files/2768886/consoleText.txt)

**Comment by jaymode** [comment]
Also hit this on a PR build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-1/4982/

```
./gradlew :server:integTest -Dtests.seed=39254641EF85F57 -Dtests.class=org.elasticsearch.action.bulk.BulkWithUpdatesIT -Dtests.method="testFailedRequestsOnClosedIndex" -Dtests.security.manager=true -Dtests.locale=en-NZ -Dtests.timezone=SystemV/MST7 -Dcompiler.java=11 -Druntime.java=8
```

**Comment by tlrx** [comment]
Closed by 29d3a708da0623b1ee29c07db99b6db49ba9089b
